packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart

```diff
@@ -0,0 +1,182 @@
+///
+/// This class implements the `package:image_picker` functionality for the web.
+class ImagePickerPlugin extends ImagePickerPlatform {
+  final ImagePickerPluginTestOverrides _overrides;
+  bool get _hasOverrides => _overrides != null;
+
+  html.Element _target;
+
+  /// A constructor that allows tests to override the function that creates file inputs.
+  ImagePickerPlugin({
+    @visibleForTesting ImagePickerPluginTestOverrides overrides,
+  }) : _overrides = overrides {
+    _target = _initTarget(_kImagePickerInputsDomId);
+  }
+
+  /// Registers this class as the default instance of [ImagePickerPlatform].
+  static void registerWith(Registrar registrar) {
+    ImagePickerPlatform.instance = ImagePickerPlugin();
+  }
+
+  @override
+  Future<PickedFile> pickImage({
+    @required ImageSource source,
+    double maxWidth,
+    double maxHeight,
+    int imageQuality,
+    CameraDevice preferredCameraDevice = CameraDevice.rear,
+  }) {
+    String capture = computeCaptureAttribute(source, preferredCameraDevice);
+    return pickFile(accept: _kAcceptImageMimeType, capture: capture);
+  }
+
+  @override
+  Future<PickedFile> pickVideo({
+    @required ImageSource source,
+    CameraDevice preferredCameraDevice = CameraDevice.rear,
+    Duration maxDuration,
+  }) {
+    String capture = computeCaptureAttribute(source, preferredCameraDevice);
+    return pickFile(accept: _kAcceptVideoMimeType, capture: capture);
+  }
+
+  /// Injects a file input with the specified accept+capture attributes, and
+  /// returns the PickedFile that the user selected locally.
+  ///
+  /// `capture` is only supported in mobile browsers.
+  /// See https://caniuse.com/#feat=html-media-capture
+  @visibleForTesting
+  Future<PickedFile> pickFile({
+    String accept,
+    String capture,
+  }) {
+    html.FileUploadInputElement input = createInputElement(accept, capture);
+    _injectAndActivate(input);
+    return _getSelectedFile(input);
+  }
+
+  // DOM methods
+
+  /// Converts plugin configuration into a proper value for the `capture` attribute.
+  ///
+  /// See: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#capture
+  @visibleForTesting
+  String computeCaptureAttribute(ImageSource source, CameraDevice device) {
+    String capture;
+    if (source == ImageSource.camera) {
+      capture = device == CameraDevice.front ? 'user' : 'environment';
+    }
+    return capture;
+  }
+
+  html.File _getFileFromInput(html.FileUploadInputElement input) {
+    if (_hasOverrides) {
+      return _overrides.getFileFromInput(input);
+    }
+    return input.files[0];
+  }
+
+  /// Handles the OnChange event from a FileUploadInputElement object
+  /// Returns the objectURL of the selected file.
+  String _handleOnChangeEvent(html.Event event) {
+    // load the file...
+    final html.FileUploadInputElement input = event.target;
+    final html.File file = _getFileFromInput(input);
+
+    if (file != null) {
```

Comment:
Is it possible to get null? If there are no files, you'll get `RangeError` I think.

Reply:
AFAIK the browser prevents you from "accepting" the dialog without selecting any files, and if the user "cancels" out, you don't even get an event (that's why the DOM is rebuilt on each "click", so the state of the picker gets fully reset in case the user didn't pick a file before.)

Reply:
Right, but if I'm reading `_getFileFromInput` correctly, if `input.files` is empty, the method will throw `RangeError`. Maybe `_getFileFromInput` should be changed to not throw, and return null instead?

Reply:
Ahhh gotcha, let me add some `?`s to `_getFileFromInput`